packages/mui-base/src/MenuUnstyled/MenuUnstyled.test.tsx

```diff
@@ -37,6 +38,39 @@ describe('MenuUnstyled', () => {
     skip: ['reactTestRenderer', 'propsSpread', 'componentProp', 'slotsProp'],
   }));
 
+  describe('after initialization', () => {
+    const spyFocus = spy();
+
+    function Test() {
+      React.useEffect(() => {
+        document.addEventListener('focus', spyFocus, true);
+        return () => {
+          document.removeEventListener('focus', spyFocus, true);
+        };
+      }, []);
+
+      return (
+        <MenuUnstyled {...defaultProps}>
+          <MenuItemUnstyled data-testid="item-1">1</MenuItemUnstyled>
+          <MenuItemUnstyled data-testid="item-2">2</MenuItemUnstyled>
+          <MenuItemUnstyled data-testid="item-3">3</MenuItemUnstyled>
+        </MenuUnstyled>
+      );
+    }
+
+    it('when menu is opened it highlights one item and it must be the first one', () => {
+      const { getAllByTestId } = render(<Test />);
+      const [firstItem, ...otherItems] = getAllByTestId(/^item-/);
+
+      expect(firstItem.tabIndex).to.equal(0);
+      otherItems.forEach((item) => {
+        expect(item.tabIndex).to.equal(-1);
+      });
+      expect(spyFocus.callCount).to.equal(1);
+      expect(spyFocus.firstCall.args[0]).to.have.property('target', firstItem);
```

Comment:
It should be sufficient to verify if `document.activeElement` is equal to `firstItem`. There's not need to introduce a spy.

Reply:
Hey @michaldudak  thanks, `document.activeElement` will still work without the fix, because Menu will always highlight the first element [MenuUnstyled](https://github.com/mui/material-ui/blob/1b6cc8ad94b9bfb295305480c6d0c1d61e0e14eb/packages/mui-base/src/MenuUnstyled/useMenu.ts#L101) but behind you don't have history what was happening after opening the Menu that's why i put a spy on focus to ensure that is called only once.

Reply:
All items at the same time they are calling the registerItem [useMenuItem:registerItem](https://github.com/mui/material-ui/blob/1b6cc8ad94b9bfb295305480c6d0c1d61e0e14eb/packages/mui-base/src/MenuItemUnstyled/useMenuItem.ts#L27) at that moment the options list is empty, that's why they got highlighted.

**Example:**
```
item1 => registerItem => highlight => focus
item2 => registerItem => highlight => focus
item3 => registerItem => highlight => focus
```
after render, effect open is called
`highlightFirstItem => item1 => focus`

Reply:
All right, good point, thanks!